src/main/java/org/folio/circulation/domain/Item.java

```diff
@@ -145,17 +156,33 @@ public String getInstanceId() {
   }
 
   public String getCallNumber() {
-    return getProperty(holdingRepresentation, "callNumber");
+    return getEffectiveCallNumberValues(CALL_NUMBER_ID);
   }
 
   public String getCallNumberPrefix() {
-    return getProperty(holdingRepresentation, "callNumberPrefix");
+    return getEffectiveCallNumberValues(CALL_NUMBER_PREFIX_ID);
   }
 
   public String getCallNumberSuffix() {
-    return getProperty(holdingRepresentation, "callNumberSuffix");
+    return getEffectiveCallNumberValues(CALL_NUMBER_SUFFIX_ID);
+  }
+
+  private String getEffectiveCallNumberValues(String propertyName) {
+    return hasItemRepresentationCallNumber()
+      ? getProperty(itemRepresentation, getItemCallNumberValueIdByPropertyName(propertyName))
+      : getProperty(holdingRepresentation, propertyName);
+  }
+
+  private boolean hasItemRepresentationCallNumber() {
+    return StringUtils.isNotBlank(getProperty(itemRepresentation, ITEM_CALL_NUMBER_ID));
   }
 
+  private String getItemCallNumberValueIdByPropertyName(String propertyName) {
```

Comment:
Can you help me understand what this does?

Does it determine the matching item property name for a given holdings property name?

Reply:
We have different keys in the JSON for the item (`itemLevelCallNumber`, `itemLevelCallNumberPrefix`) and the holding (`callNumber`).  Basically, it determines the corresponding `callNumbers` property name for a given holding property name.

Reply:
Thanks for helping me understand. Maybe the method and parameter could be renamed to help with that.

What do you think of the suggestion below?

```suggestion
  private String mapToItemCallNumberPropertyName(String holdingsPropertyName) {
```

Reply:
Thanks for the comment, I corrected the name of the method.